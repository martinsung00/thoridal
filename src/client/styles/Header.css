.root {
  width: 100%;
  height: 8.5%;
  margin-bottom: 5%;
  top: 0;
  left: 0;
  position: absolute;
  background-color: rgb(196, 196, 196);
  color: black;
  transition: top 0.3s;
}

.title {
  position: relative;
  float: left;
  font-family: Helvetica, Arial, sans-serif;
  font-weight: bold;
  font-size: 27px;
  margin-left: 43%;
  margin-top: 1%;
}

.mainLogo {
  position: relative;
  float: left;
  margin-top: 1.2%;
  margin-left: 1%;
  height: 25px;
  width: 25px;
}

.searchIcon {
  position: relative;
  float: left;
  margin-top: -0.5%;
  margin-left: 2%;
  padding-right: 0%;
  height: 30px;
  width: 30px;
  cursor: pointer;
}

.searchbar {
  float: right;
  position: relative;
  margin-left: 90%;
  margin-right: 5%;
  margin-top: -2.3%;
  background-color: rgba(255, 255, 255, 0.952);
  border-radius: 2.5em;
  width: 16.5em;
  height: 1.8em;
}

.search {
  position: relative;
  background-color: rgba(255, 255, 255, 0.952);
  margin-top: 2%;
  margin-left: 6.8%;
  border: none;
  height: 1.3em;
  width: 5.5em;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 16px;
  outline: none;
}

.searchType {
  position: relative;
  float: right;
  background-color: rgba(255, 255, 255, 0.952);
  margin-top: -8.1%;
  margin-right: 21%;
  padding-left: 3%;
  padding-right: -1%;
  border-left: 0.7px solid darkgray;
  width: 2.5em;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 15px;
  outline: none;
}

.searchBy {
  border: none;
  font-size: 14.5px;
  font-weight: bold;
  color: rgb(116, 116, 116);
  background-color: white;
}
