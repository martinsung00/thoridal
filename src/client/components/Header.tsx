import React from "react";
import styles from "../styles/Header.css";
import logo from "../logo/logo.svg";
import spyglass from "../logo/spyglass.svg";

interface Props {
  // To-do: Specify prop types
}

interface State {
  // To-do: Add more state types
  searchType: string;
  searchTerm: string | undefined;
}

export default class Header extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      searchType: "id",
      searchTerm: "",
    };
  }

  _handleChange(event: React.ChangeEvent<HTMLInputElement>): void {
    this.setState({
      searchTerm: event.target.value,
    });
  }

  _handleSelect(event: React.FormEvent<HTMLSelectElement>) {
    this.setState({
      searchType: event.currentTarget.value,
    });
  }

  render() {
    return (
      <div>
        <div className={styles.head}>
          <div className={styles.title}>THORIDAL</div>
          <img src={logo} alt="Main Logo" className={styles.mainLogo} />
          <div className={styles.searchbar}>
            <img
              src={spyglass}
              alt="Search Icon"
              className={styles.searchIcon}
            />
            <div>
              <input
                type="text"
                placeholder="Search By"
                value={this.state.searchTerm}
                onChange={this._handleChange.bind(this)}
                className={styles.search}
              />
            </div>
            <div className={styles.searchType}>
              <select
                className={styles.sortBy}
                value={this.state.searchType}
                onChange={this._handleSelect.bind(this)}
                defaultValue="id"
              >
                <option value="id">ID</option>
                <option value="company">Company</option>
                <option value="ticker">Ticker</option>
                <option value="date">Date</option>
              </select>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
