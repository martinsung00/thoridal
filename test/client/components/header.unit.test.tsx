import React from "react";
import renderer from "react-test-renderer";
import Header from "../../../src/client/components/Header";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("App Tests", function () {
  it("should render header", function () {
    const container = renderer.create(<Header />).toJSON();

    expect(container).toMatchSnapshot();
  });

  it("should change the current state when typing in the search bar", async function () {
    const wrapper = shallow(<Header />);
    const event = {
      preventDefault() {},
      target: { value: "Hello World" },
    };

    wrapper.setState({
      searchType: "id",
      searchTerm: "",
    });

    wrapper.find("input").simulate("change", event);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state()).toEqual({
      searchType: "id",
      searchTerm: "Hello World",
    });

    wrapper.unmount();
  });

  it("should change the current state when selecting search types", async function () {
    const wrapper = shallow(<Header />);
    const event = {
      preventDefault() {},
      currentTarget: { value: "company" },
    };

    wrapper.setState({
      searchType: "id",
      searchTerm: "",
    });

    wrapper.find("select").simulate("change", event);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state()).toEqual({
      searchType: "company",
      searchTerm: "",
    });

    wrapper.unmount();
  });
});
