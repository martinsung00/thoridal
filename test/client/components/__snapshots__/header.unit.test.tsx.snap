// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests should render header 1`] = `
<div>
  <div>
    <div>
      THORIDAL
    </div>
    <img
      alt="Main Logo"
      src={[Function]}
    />
  </div>
</div>
`;
